core/src/Kokkos_Parallel.hpp

```diff
@@ -124,6 +124,11 @@ struct FunctorPolicyExecutionSpace<
   typedef typename Functor::execution_space execution_space;
 };
 
+template<typename Policy>
+struct PolicyDeviceId {
+    constexpr static int value = Policy::device_id::value;
```

Comment:
There should be a defaulted version of this that returns some value if `device_id` isn't present.  We don't want to make `device_id` a mandatory part of the `ExecutionPolicy` concept just to facilitate this.